actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -240,6 +288,56 @@ def handle_unverified_request
       end
     end
 
+    class SessionStore
+      def fetch(request)
+        request.session[:_csrf_token]
+      end
+
+      def store(request, csrf_token)
+        request.session[:_csrf_token] = csrf_token
+      end
+
+      def reset(request)
+        request.session.delete(:_csrf_token)
+      end
+    end
+
+    class CookieStore
+      def initialize(cookie = :csrf_token)
+        @cookie = cookie
```

Comment:
This would be clearer I think:
```suggestion
        @cookie_name = name
```